src/NodeAutoCompleteViewExtension/ViewModels/NodeAutoCompleteBarViewModel.cs

```diff
@@ -802,16 +810,26 @@ internal void AddCluster(ClusterResultItem ClusterResultItem)
 
             List<List<NodeItem>> nodeStacks = NodeAutoCompleteUtilities.ComputeNodePlacementHeuristics(clusterConnections, clusterNodes);
 
+            //store our nodes and wires to allow for one undo
+            List<ModelBase> newNodesAndWires = new List<ModelBase>();
+
             double xoffset = node.X + node.NodeModel.Width;
             foreach (var nodeStack in nodeStacks)
             {
                 xoffset += node.NodeModel.Width;
                 foreach(var newNode in nodeStack)
                 {
-                    // Retreive assembly name and node full name from type.id.
+                    // Retrieve assembly name and node full name from type.id.
                     var typeInfo = wsViewModel.NodeAutoCompleteSearchViewModel.GetInfoFromTypeId(newNode.Type.Id);
                     dynamoViewModel.Model.ExecuteCommand(new DynamoModel.CreateNodeCommand(Guid.NewGuid().ToString(), typeInfo.FullName, xoffset, node.NodeModel.Y, false, false));
+
+                    //disallow the node creation command from the undo group, we group node creation and wires below
+                    wsViewModel.Model.UndoRecorder.PopFromUndoGroup();
```

Comment:
oh interesting, I guess with this we can keep the current implementation with commands?

Reply:
Yes! :)